### PR TITLE
comui-4546 add bulk get to channel reciever

### PR DESCRIPTION
**Pull request by rhkwong**

### CODE REVIEW

File: rdbi-recipes/pom.xml [hunk 1]
```diff
@@ -89,6 +89,11 @@
             <artifactId>awaitility</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (phutwo) [inline rdbi-recipes/pom.xml]: scope should be test only, but i'm surprised this has to be added. is this correct?

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiver.java [hunk 2]
```diff
@@ -38,6 +41,32 @@ public GetResult get(
         );
 
         @Query(
+            "local tables_size = KEYS[1]\n" +
```
Comment (phutwo) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiver.java]: change the name to number_of_channels

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiverTest.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+        for (String eachChannel : channels) {
+            handle.jedis().del(eachChannel + ":queue");
+            handle.jedis().del(eachChannel + ":depth");
+        }
+        handle.close();
+    }
+
+
+    @Test
```
Comment (phutwo) [inline rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiverTest.java]: explain why it's ignore and your results

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiver.java [hunk 4]
```diff
@@ -129,6 +179,30 @@ public GetResult get(String channel, Long lastSeenId, String copyDepthToKey) {
...
+            DAO dao = handle.attach(DAO.class);
+                List<String> depthKeys = channels.stream().map(eachChannel -> ChannelPublisher.getChannelDepthKey(eachChannel)).collect(Collectors.toList());
+                List<String>  queueKeys = channels.stream().map(eachChannel -> ChannelPublisher.getChannelQueueKey(eachChannel)).collect(Collectors.toList());
+
+                List<String> keysList = new ArrayList<>(depthKeys.size()*3+1);
+                keysList.add(String.valueOf(depthKeys.size()));
+                for (int i = 0; i < channels.size(); i++) {
+                    keysList.add(queueKeys.get(i));
+                    keysList.add(depthKeys.get(i));
+                    keysList.add("0");
```
Comment (phutwo) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiver.java]: what's this 0 for?

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiver.java [hunk 5]
```diff
@@ -38,6 +41,32 @@ public GetResult get(
...
+            "        results[#results + 1] = tostring(current_count)\n" +
+            "        bulk_result[i] = results\n" +
+            "    end\n" +
+            "end\n" +
+            "return bulk_result;"
+        )
+        @Mapper(GetBulkResultMapper.class)
+        public GetBulkResult getMultiV2(
+                @BindKey("allKeys") List<String> inputKeys,
+                @BindArg("allArgs") List<String> inputArgs
```
Comment (phutwo) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiver.java]: woah whoa, let's talk, you're suppose to use the args list, the keys list is only to tell redis what keys you'll be touching, the arg list should contain the depths etc.

File: rdbi-recipes/pom.xml (hunk 1 above)
Comment (rhkwong) [reply rdbi-recipes/pom.xml]: i think so....i need junit 4 for the @ignore annotation.

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiver.java [hunk 6]
```diff
@@ -62,6 +64,34 @@ public GetResult get(
         );
 
         @Query(
+                "local number_of_channels = ARGV[1]\n" +
+                        "local bulk_result = {}\n" +
+                        "for i = 1, number_of_channels do\n" +
+
+                        "print(i) \n" +
```
Comment (phutwo) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/channel/ChannelLuaReceiver.java]: get rid of all prints

File: pom.xml [hunk 7]
```diff
@@ -6,7 +6,7 @@
     <parent>
         <groupId>com.lithium.dbi.rdbi</groupId>
         <artifactId>rdbi-parent</artifactId>
-        <version>0.69</version>
+        <version>0.70-SNAPSHOT</version>
```
Comment (phutwo) [inline pom.xml]: lol no, when you release this, there's a way to do this

File: rdbi-recipes/pom.xml (hunk 1 above)
Comment (phutwo) [reply rdbi-recipes/pom.xml]: add <scope>test</scope>

File: pom.xml (hunk 7 above)
Comment (rhkwong) [reply pom.xml]: i think i messed up something with git

**Pull request merged**